server/src/main/java/org/elasticsearch/action/search/TransportMultiSearchAction.java

```diff
@@ -168,6 +172,9 @@ public void onResponse(final SearchResponse searchResponse) {
 
             @Override
             public void onFailure(final Exception e) {
+                if (ExceptionsHelper.status(e).getStatus() >= 500 && ExceptionsHelper.isNodeOrShardUnavailableTypeException(e) == false) {
+                    logger.warn("TransportMultiSearchAction failure for request " + request.request, e);
```

Comment:
Here the SearchRequest#toString seems more useful to print out.

Reply:
Yes, that's what we get here (toString is implicit). Example output from manual testing:

```
TransportMultiSearchAction failure for request SearchRequest{searchType=QUERY_THEN_FETCH, indices=[blogs], indicesOptions=IndicesOptions[ignore_unavailable=false, allow_no_indices=true, expand_wildcards_open=true, expand_wildcards_closed=false, expand_wildcards_hidden=false, allow_aliases_to_multiple_indices=true, forbid_closed_indices=true, ignore_aliases=false, ignore_throttled=true], routing='null', preference='null', requestCache=null, scroll=null, maxConcurrentShardRequests=0, batchedReduceSize=512, preFilterShardSize=null, allowPartialSearchResults=true, localClusterAlias=null, getOrCreateAbsoluteStartMillis=-1, ccsMinimizeRoundtrips=true, source={"from":0,"size":10,"query":{"match":{"message":{"query":"hello world"}}},"explain":false}}
```